test/API/H5_api_dataset_test.c

```diff
@@ -2711,19 +3127,23 @@ test_create_dataset_creation_properties(void)
 error:
     H5E_BEGIN_TRY
     {
-        H5Sclose(compact_fspace_id);
-        H5Sclose(fspace_id);
-        H5Tclose(compact_dtype);
-        H5Tclose(dset_dtype);
-        H5Dclose(dset_id);
-        H5Pclose(dcpl_id);
-        H5Gclose(group_id);
-        H5Gclose(container_group);
-        H5Fclose(file_id);
-    }
-    H5E_END_TRY
+        if (read_buf) {
```

Comment:
Looks like an extra `{` got inserted here, causing all the frees below to only be called if read_buf is non-NULL here.